### PR TITLE
Update configs to use Ruff format instead of Black

### PR DESCRIPTION
**Pull request by Avasam**
Follow-up to #2340 (closes #2340 if this is merged first)

Ruff format's style is based on Black's, but with a handful of improvements. It also runs much faster on the entire repo, which is nice locally. There's an added advantage of being mostly Python-version agnostic (at least we don't need to run it in a Python environment, so it won't drop support for older versions faster than we do, unlike Black). Since we already use Ruff's linter, this doesn't add any new tool (and actually removes one) nor configuration file.

First commit are manual changes. Second commit is only running `ruff format`

### CODE REVIEW

File: .editorconfig [hunk 1]
```diff
@@ -9,7 +9,7 @@ indent_size = 4
 max_line_length = 120 # Same as .clang-format
 
 [*.py]
-max_line_length = 88 # Same as Black
+max_line_length = 88 # Same as Ruff's default
```
Comment (Avasam) [inline .editorconfig:12]: If you'd like to change that to have a consistent line-length across all files. I can change it in `.clang-format` and/or `ruff.toml` to your preference in an follow-up PR.

Comment (mhammond) [reply .editorconfig:12]: > If you'd like to have a consistent line-length across all files. 

God no - that's something I wish these linting tools cared less about :)

**Review by mhammond — APPROVED** [review]
Thanks! Could you please arrange for the auto-formatted stuff to land in its own commit then add that commit to https://github.com/mhammond/pywin32/blob/main/.git-blame-ignore-revs

**Comment by Avasam** [comment]
> Thanks! Could you please arrange for the auto-formatted stuff to land in its own commit then add that commit to [`main`/.git-blame-ignore-revs](https://github.com/mhammond/pywin32/blob/main/.git-blame-ignore-revs?rgh-link-date=2024-10-14T00%3A13%3A43Z)

Done !

**Pull request merged**